pkg/core/core.go

```diff
@@ -315,3 +337,84 @@ func (c *Core) resetNsSwitchConfig() error {
 	c.logger.Debug("Successfully reset the nsswitch config of linux")
 	return nil
 }
+
+const (
+	keployTestPort   = 56789
+	keployRecordPort = 36789
+)
+
+func (c *Core) setUpReplayTesting(ctx context.Context) error {
+	setUpErr := errors.New("failed to setup the keploy replay testing")
+
+	keployRecordPid, err := utils.GetPIDByPort(ctx, c.logger, keployRecordPort)
+	if err != nil {
+		c.logger.Error("failed to get the keployRecord pid", zap.Error(err))
+		utils.LogError(c.logger, err, "failed to get the keployRecord pid from port", zap.Any("port", keployRecordPort))
+		return setUpErr
+	}
+	c.logger.Debug(fmt.Sprintf("keployRecord pid:%v", keployRecordPid))
+
+	err = c.TransmitTestBenchKeployPIDs(0, keployRecordPid)
+	if err != nil {
+		return setUpErr
+	}
+
+	err = c.TransmitTestBenchKeployPorts(0, uint32(keployRecordPort))
+	if err != nil {
+		return setUpErr
+	}
+
+	err = c.TransmitTestBenchKeployPorts(1, uint32(keployTestPort))
+	if err != nil {
+		return setUpErr
+	}
+
+	// to get the pid of keployTest binary in keployRecord binary, we have to wait for some time till the proxy server is started
+	// TODO: find other way to filter child process (keployTest) pid in parent process binary (keployRecord)
+	time.Sleep(10 * time.Second) // just for test bench.
+
+	return nil
+}
+
+func (c *Core) setUpRecordTesting(ctx context.Context) error {
+
+	go func() {
+		timeout := 30 * time.Second
+		startTime := time.Now()
+
+		ticker := time.NewTicker(500 * time.Millisecond)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				keployTestPid, err := utils.GetPIDByPort(ctx, c.logger, keployTestPort)
+				if err != nil {
+					c.logger.Debug("failed to get the keploytest pid", zap.Error(err))
+					continue
+				}
+
+				if keployTestPid == 0 {
+					continue
+				}
+
+				c.logger.Debug("keploytest pid", zap.Uint32("pid", keployTestPid))
+
+				// sending keploytest binary pid in keployrecord binary to filter out ingress/egress calls related to keploytest binary.
+				_ = c.Hooks.TransmitTestBenchKeployPIDs(1, keployTestPid)
+
+				return
+
+			case <-time.After(timeout - time.Since(startTime)):
+				c.logger.Debug("Timeout reached, exiting loop from setupRecordTesting")
```

Comment:
Is the debug log sufficient? what is the consequence for the user not getting any warn or error log here?